compiler/rustc_mir/src/transform/const_prop.rs

```diff
@@ -89,6 +89,10 @@ impl<'tcx> MirPass<'tcx> for ConstProp {
             return;
         }
 
+        if !tcx.consider_optimizing(|| format!("ConstantPropagation {:?} {:?}", def_id, hir_id)) {
```

Comment:
I think this is basically the same feedback @oli-obk gave but we should try to make this as fine-grained as possible. The primary use for optimization fuel is debugging mis-optimizations. To do that effectively, we need to be able to bisect a compilation session down to the exact incorrect optimization that occurs. Doing this at the pass level is too broad. We really need to check this each time an individual optimization operation is applied.

Reply:
Does it make more sense to put the check in [`should_const_prop`](https://github.com/rust-lang/rust/blob/master/compiler/rustc_mir/src/transform/const_prop.rs#L886) then?

Reply:
Yes, that makes sense to me 👍